src/rustc-driver/remarks-on-perma-unstable-features.md

```diff
@@ -0,0 +1,52 @@
+## Requirements for `rustc_private`
+
+### Overview
+
+The `rustc-private` feature allows external crates to use compiler internals, but requires specific components to be properly installed. This document explains how to resolve common linker errors related to missing LLVM libraries when working with this feature.
```

Comment:
Nit:

```suggestion
The `rustc_private` feature allows external crates to use compiler internals.
```

(The requirement bits are incomplete here and redundant with the following bits)